ddtrace/internal/writer.py

```diff
@@ -532,7 +533,10 @@ def write(self, spans=None):
                     # are initialized
                     if asbool(os.getenv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", True)):
                         telemetry_writer.enable()
-                    # appsec remote config should be enabled/started after the global tracer and configs
+                    # remote config should be enabled/started after the global tracer and configs are initialized
+                    if asbool(os.environ.get("DD_REMOTE_CONFIGURATION_ENABLED", "true")):
```

Comment:
I think this implies that DI without a tracer won't work, so this change has the potential of breaking DI.

Reply:
We need this approach to fix the gevent problem with RC, we update too Debugger part, isn't it enough to work with Debugger too? What do you propose?

Reply:
@P403n1x87 Is this a blocker? Can we verify whether DI breaks with this change?

Reply:
@P403n1x87  Do you check this change? :point_right:  https://github.com/DataDog/dd-trace-py/pull/5088/files#diff-d14a7422eb9f2743a3db00be1dbc2245ca0706fc06c1120b5458516af389322fR251

All Debugger tests work fine with that change

Reply:
Ah no I missed that, but I think it unfortunately creates more problems than it solves:
- The way of enabling DI is with `DD_DYNAMIC_INSTRUMENTATION_ENABLED`; having to set another variable to enable DI is not the behaviour that we want to support.
- This might solve the issue for ASM, but if someone enables DI then we would have the same problem ASM is trying to fix.

Reply:
You don't need to add another environment variable, `DD_REMOTE_CONFIGURATION_ENABLED` is True by default but we need this environment variable to run tests isolated.



> * This might solve the issue for ASM, but if someone enables DI then we would have the same problem ASM is trying to fix.

Bugs one by one :stuck_out_tongue: We fixed ASM/APM in this PR, we could merge this PR and #5105 fix this bug in DI, right? :+1: